### DISCUSSION TITLE
How to perform the following command using gtag ?

### DISCUSSION THREAD

**Discussion by patrickmonteiro** [johannschopplich/nuxt-gtag#41] [General]
I need to run:

`dataLayer.push({ ecommerce: null });`

https://developers.google.com/analytics/devguides/collection/ua/gtm/enhanced-ecommerce?hl=pt-br#clear-ecommerce

But I still don't understand how to do this with gtag. Would it be an event?

**Comment by johannschopplich** [comment]
To add values to the `dataLayer` property of the global window scope, you can use [`useGtag`](https://github.com/johannschopplich/nuxt-gtag#usegtag) as follows:

```ts
// Can be used on the server and client-side, but will only have an effect in the client
const { gtag } = useGtag()
gtag({ ecommerce: null })
```

Under the hood, the extracted `gtag` function will add properties to `dataLayer`. For reference, see: https://github.com/johannschopplich/nuxt-gtag/blob/1af82b9f0ec24f7891d778e14225e90a3f8c329d/src/runtime/gtag.ts#L2

I hope that helps you going forward.
